tests/src/tests/command_test_utils.go

```diff
@@ -0,0 +1,256 @@
+	} else {
+		execCommand.WorkingDir = ""
+	}
+
+	execCommand.HotReloadCapable = GetBinaryDecision()
+	LogMessage(fmt.Sprintf("   ....... HotReloadCapable: %t", execCommand.HotReloadCapable))
+
+	if GetBinaryDecision() {
+		execCommand.Env = AddEnv(GetRandomNumber(4))
+	} else {
+		execCommand.Env = nil
+	}
+
+}
+
+func (devfile TestDevfile) replaceSchemaCommand(command schema.Command) {
+	for i := 0; i < len(devfile.SchemaDevFile.Commands); i++ {
+		if devfile.SchemaDevFile.Commands[i].Id == command.Id {
+			devfile.SchemaDevFile.Commands[i] = command
+			break
+		}
+	}
+}
+
+func getSchemaCommand(commands []schema.Command, id string) (*schema.Command, bool) {
+	found := false
+	var schemaCommand schema.Command
+	for _, command := range commands {
+		if command.Id == id {
+			schemaCommand = command
+			found = true
+			break
+		}
+	}
+	return &schemaCommand, found
+}
+
+func createCompositeCommand() *schema.CompositeCommand {
+
+	compositeCommand := schema.CompositeCommand{}
+
+	setCompositeCommandValues(&compositeCommand)
+
+	return &compositeCommand
+}
+
+func setCompositeCommandValues(compositeCommand *schema.CompositeCommand) {
+	numCommands := GetRandomNumber(3)
+
+	compositeCommand.Commands = make([]string, numCommands)
+	for i := 0; i < numCommands; i++ {
+		compositeCommand.Commands[i] = GetRandomUniqueString(8, false)
+		LogMessage(fmt.Sprintf("   ....... command %d of %d : %s", i, numCommands, compositeCommand.Commands[i]))
+	}
+
+	if GetRandomDecision(2, 1) {
+		compositeCommand.Group = addGroup()
+	}
+
+	if GetBinaryDecision() {
+		compositeCommand.Label = GetRandomString(12, false)
+		LogMessage(fmt.Sprintf("   ....... label: %s", compositeCommand.Label))
+	}
+
+	if GetBinaryDecision() {
+		compositeCommand.Parallel = true
+		LogMessage(fmt.Sprintf("   ....... Parallel: %t", compositeCommand.Parallel))
+	}
+}
+
+func (devfile TestDevfile) VerifyCommands(parserCommands []schema.Command) error {
+
+	LogMessage("Enter VerifyCommands")
+	errorString := ""
+
+	// Compare entire array of commands
+	if !cmp.Equal(parserCommands, devfile.SchemaDevFile.Commands) {
+		errorString += LogMessage(fmt.Sprintf(" --> ERROR: Command array compare failed."))
+		// Array compare failed. Narrow down by comparing indivdual commands
+		for _, command := range parserCommands {
+			if testCommand, found := getSchemaCommand(devfile.SchemaDevFile.Commands, command.Id); found {
+				if !cmp.Equal(command, *testCommand) {
+					parserFilename := AddSuffixToFileName(devfile.FileName, "_"+command.Id+"_Parser")
+					testFilename := AddSuffixToFileName(devfile.FileName, "_"+command.Id+"_Test")
+					LogMessage(fmt.Sprintf("   .......marshall and write devfile %s", devfile.FileName))
+					c, err := yaml.Marshal(command)
+					if err == nil {
+						err = ioutil.WriteFile(parserFilename, c, 0644)
+					}
+					LogMessage(fmt.Sprintf("   .......marshall and write devfile %s", devfile.FileName))
```

Comment:
for the log messages format, e.g. `   .......` and ` --> ERROR:` have to be manually passes in. suggest separate functions:
```
LogErrorMessage(){
testLogger.Println("--> ERROR: " + message)
}
```
and
```
LogInfoMessage(){
testLogger.Println("   ....... " + message)
}
```

Reply:
Sure, if that is a format you prefer, also its a minor improvement so I prefer to add it in part 2.

Reply:
Can you also create an issue to track this as well? I just do not think it's a good idea to manually input the log format for every log message.